src/it/gh-182/src/main/java/com/y/BClass.java

```diff
@@ -0,0 +1,32 @@
+package com.y;
+
+import com.AnnotationA;
+import java.util.ArrayList;
+import java.util.List;
+
+public class BClass {
+    public List<java.util.@com.AnnotationA List<Long>> fooField;
+    public List<@com.AnnotationA List<Long>> barField;
+    public List<java.util.@AnnotationA List<Long>> bazField;
+    public List<@AnnotationA List<Long>> borkField;
+
+    public List<java.util.@com.AnnotationA List<Long>> foo() { return null; }
+    public List<@com.AnnotationA List<Long>> bar() { return null; }
+    public List<java.util.@AnnotationA List<Long>> baz() { return null; }
+    public List<@AnnotationA List<Long>> bork() { return null; }
+
+    public BClass.Container<com.y.BClass.@com.AnnotationA Container<Long>> eggs() { return null; }
```

Comment:
Simulated the permutations of the more funky cases that annotation processors and more complex code can generate.

This is valid for Java 8 and newer, but breaks the QDox parser.